openssl/src/x509/extension.rs

```diff
@@ -428,6 +443,258 @@ impl AuthorityKeyIdentifier {
     }
 }
 
+#[cfg(ossl110)]
+pub struct SbgpAsIdentifier {
+    critical: bool,
+    asn: Vec<(u32, u32)>,
+    inherit: bool,
```

Comment:
I think this is a good idea. However I think patterns like type-state would be overkill in this scenario, so assertions / panics remain nessecary to prevent silent failures.